run.py

```diff
@@ -108,39 +122,58 @@
       scores = []
 
       inputFileName = str(os.path.join(submissionBuildDir, "221575.pgm"))
-      runCmd = "./canny " + inputFileName + " 0.5 0.7 0.9 2>&1"
+      if sys.platform != 'win32':      
+        runCmd = "./canny " + inputFileName + " 0.5 0.7 0.9 2>&1"
+      else:
+        runCmd = "canny.exe " + inputFileName + " 0.5 0.7 0.9"
       valid = True
       try:
-        subprocess.check_call(runCmd, shell=True, stdout=FNULL, stderr=FNULL)
+        subprocess.check_call(runCmd, shell=True, stdout=FNULL, stderr=FNULL)   
       except:
         valid = False
 
       if valid:
         print("  run - OK")
       if not valid:
         print("  run - Failed")
-      
+        print("  input file:" + inputFileName)
+     
       outputFileName = str(os.path.join(submissionBuildDir, "221575.pgm_s_0.50_l_0.70_h_0.90.pgm"))
       goldenFileName = str(os.path.join(saveCwd, "221575-out-golden.pgm"))
 
       valid = True
       try:
-        subprocess.check_call("./validate " + goldenFileName + " " + outputFileName, shell=True, stdout=FNULL, stderr=FNULL)
+        if sys.platform != 'win32':            
+          subprocess.check_call("./validate " + goldenFileName + " " + outputFileName, shell=True, stdout=FNULL, stderr=FNULL)
+        else:
+          subprocess.check_call("validate.exe " + goldenFileName + " " + outputFileName, shell=True, stdout=FNULL, stderr=FNULL)          
       except:
         valid = False
      
       if valid:
         print("  validation - OK")
       if not valid:
         print("  validation - Failed")
+        print("  validation - validate.exe " + goldenFileName + " " + outputFileName)
 
+      if verbose:
+        print ("Running 10 measurements loop ...")
+        
       for x in range(0, 10):
-        output = subprocess.check_output("time -p " + runCmd, shell=True) 
-        for row in output.split(b'\n'):
-          if b'real' in row:
-            real, time = row.split(b' ')
-            scores.append(float(time))
-
+        if sys.platform != 'win32':
+          output = subprocess.check_output("time -p " + runCmd, shell=True)
+          for row in output.split(b'\n'):
+            if b'real' in row:
+              real, time = row.split(b' ')
+              scores.append(float(time))          
+        else:
+          print("  run - WIN 32")
+          output = subprocess.check_output("powershell -Command \"Measure-Command {" + runCmd + " | Out-Default}\" | findstr TotalSeconds" , shell=True)
```

Comment:
Do you know if powershell is available by default?

Reply:
Docs says so:

"Windows PowerShell comes installed by default in every Windows, starting with Windows 7 SP1 and Windows Server 2008 R2 SP1."

https://docs.microsoft.com/de-de/powershell/scripting/windows-powershell/install/installing-windows-powershell?view=powershell-7.1

Reply:
Ok, thanks!